packages/astro/src/assets/utils/remoteProbe.ts

```diff
@@ -2,6 +2,13 @@ import { AstroError, AstroErrorData } from '../../core/errors/index.js';
 import type { ImageMetadata } from '../types.js';
 import { imageMetadata } from './metadata.js';
 
+/**
+ * Infers the dimensions of a remote image by streaming its data and analyzing it progressively until sufficient metadata is available.
+ *
+ * @param {string} url - The URL of the remote image from which to infer size metadata.
+ * @return {Promise<Omit<ImageMetadata, 'src' | 'fsPath'>>} Returns a promise that resolves to an object containing the image dimensions metadata excluding `src` and `fsPath`. Throws an error if the image dimensions cannot be determined or the fetch fails.
+ * @throws {AstroError} Thrown when the fetching fails or metadata cannot be extracted.
```

Comment:
Isn't it redundant to have `Throws` explained in `@return` in addition to `@throws`?
```suggestion
 * @return {Promise<Omit<ImageMetadata, 'src' | 'fsPath'>>} Returns a promise that resolves to an object containing the image dimensions metadata excluding `src` and `fsPath`.
 * @throws {AstroError} Thrown when the fetching fails or metadata cannot be extracted.
```